### PR TITLE
CHE-11496: Search container in the all workspace pods.

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
### What issues does this PR fix or reference:
https://github.com/eclipse/che/issues/11496

Signed-off-by: Oleksandr Andriienko <oandriie@redhat.com>

### CODE REVIEW

**Comment by skabashnyuk** [comment]
is there any chance to have a test here?

**Comment by AndrienkoAleksandr** [comment]
@skabashnyuk I create issue for that https://github.com/eclipse/che-machine-exec/issues/8.

**Comment by garagatyi** [comment]
Please, test this PR and comment when you are sure that it works

File: exec/kubernetes-infra/machine_filter.go [hunk 1]
```diff
@@ -46,22 +47,15 @@ func findMachineContainerInfo(execManager KubernetesExecManager, identifier *mod
...
 	var containerName string
-	for _, container := range containers {
-		for _, env := range container.Env {
-			if env.Name == MachineName && env.Value == identifier.MachineName {
-				containerName = container.Name
-			}
+	var pod v1.Pod
+	for _, pod = range pods.Items {
+		if containerName = findContainerName(pod, identifier.MachineName); containerName != "" {
+			continue
```
Comment (garagatyi) [inline exec/kubernetes-infra/machine_filter.go]: We can move return here and simplify the code

Comment (AndrienkoAleksandr) [reply exec/kubernetes-infra/machine_filter.go]: Yes, thanks

**Comment by AndrienkoAleksandr** [comment]
@garagatyi  I completed manual testing. @skabashnyuk I applied tests.

**Comment by AndrienkoAleksandr** [comment]
@benoitf Do we need CQ in case if I used auto-generated mocks in the code ?

**Comment by AndrienkoAleksandr** [comment]
@skabashnyuk testify uses one more dependencies https://github.com/ws-skeleton/che-machine-exec/pull/3/commits/f75c557a36285f3dd06ecce54a77e17e357bd3c0#diff-836546cc53507f6b2d581088903b1785R110  it's from here https://github.com/stretchr/testify/blob/master/Gopkg.toml#L15

**Pull request merged**